astroquery/jplhorizons/core.py

```diff
@@ -1032,17 +1041,18 @@ def vectors_async(self, *, get_query_payload=False,
 
         URL = conf.horizons_server
 
-        # check for required information
+        # check for required information and assemble commandline stub
         if self.id is None:
             raise ValueError("'id' parameter not set. Query aborted.")
+        elif isinstance(self.id, dict):
+            commandline = f"g:{self.id['lon']},{self.id['lat']},{self.id['elevation']}@{self.id['body']}"
```

Comment:
Usually I prefer f-strings over `str.format()`, but in this case the following can be considered:
```suggestion
            commandline = "g:{lon},{lat},{elevation}@{body}".format(**self.id)
```

Reply:
this is prettier, yes.